app/models/user.rb

```diff
@@ -71,6 +71,13 @@ def set_next_email_as_primary
     default_email_record.update_attributes(primary: true)
   end
 
+  # Loads unread popups of a user.
+  #
+  # @return [Array<UserNotification>] The unread popup notifications
+  def unread_popups
+    notifications.popups.unread_by(self).order(created_at: :desc)
```

Comment:
hmm...I think I need to do something with that gem so that it can support our requirements. So maybe I'll fix this part in next pr.